plugins/webp-uploads/hooks.php

```diff
@@ -803,3 +803,64 @@ function webp_uploads_enable_additional_mime_type_support_for_all_sizes( array $
 	return $allowed_sizes;
 }
 add_filter( 'webp_uploads_image_sizes_with_additional_mime_type_support', 'webp_uploads_enable_additional_mime_type_support_for_all_sizes' );
+
+/**
+ * Converts palette PNG images to truecolor PNG images.
+ *
+ * GD cannot convert palette-based PNG to WebP/AVIF formats, causing conversion failures.
+ * This function detects and converts palette PNG to truecolor during upload.
+ *
+ * @since n.e.x.t
+ *
+ * @param array<string, mixed>|mixed $file The uploaded file data.
+ * @return array<string, mixed>|mixed The modified file data.
+ */
+function webp_uploads_convert_palette_png_to_truecolor( $file ) {
+	if ( ! is_array( $file ) ) {
+		return $file;
+	}
```

Comment:
I think it's slightly better to fix up the type of the filtered argument rather than passing through the bad value type.

```suggestion
 * @return array<string, mixed> The modified file data.
 */
function webp_uploads_convert_palette_png_to_truecolor( $file ): array {
	// Because plugins do bad things.
	if ( ! is_array( $file ) ) {
		$file = array();
	}
```

Reply:
I'm not sure about this there are conflicting reviews. 😅
See: https://github.com/WordPress/performance/pull/2024#discussion_r2217410708.

Reply:
I think you'll see my suggestion is used elsewhere in the PL plugins, and it has the benefit of being able to specify the `array` type as the return value, rather than `mixed`.